sdk/Pulumi.Tests/Resources/ComponentResourceTests.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright 2016-2023, Pulumi Corporation
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Pulumi.Testing;
+using Xunit;
+
+namespace Pulumi.Tests.Resources
+{
+    public class ComponentResourceTests
+    {
+        [Fact]
+        public async void PropagatesProviderOption()
+        {
+            var mocks = new MinimalMocks();
+            var options = new TestOptions();
+
+            await Deployment.TestAsync(mocks, options, () =>
+            {
+                var provider = new ProviderResource(
+                    "test",
+                    "prov",
+                    ResourceArgs.Empty
+                );
+
+                var component = new ComponentResource(
+                    "custom:foo:component",
+                    "comp",
+                    new ComponentResourceOptions
+                    {
+                        Provider = provider,
+                    }
+                );
+
+                var custom = new CustomResource(
+                    "test:index:MyResource",
+                    "custom",
+                    ResourceArgs.Empty,
+                    new CustomResourceOptions
+                    {
+                        Parent = component,
+                    }
+                );
+
+                Assert.Equal(provider, custom._provider);
+            });
```

Comment:
Could you add a comment to clarify the motivation of what's going on here. It makes sense at the end when you assert equal, but it's not immediately clear and you have to keep a lot of stuff in your head if you were reading it in the codebase without the PR context.

Reply:
Good point. I realize that the context from the commit/issue wasn't carried over to the code. 
Will fix.